Packs/EclecticIQ/Integrations/EclecticIQ/EclecticIQ.py

```diff
@@ -1,794 +1,679 @@
+    :param feeds: permissions_data
+    :type response: list
+        [{"id": 1, "name": "read history-events"},{"id": 2,"name": "read discovery-rules"}...]
+    :return: List of permission ids
+        [33, 59, 66,78]
+    :rtype: list
+    """
+    wanted_permissions = [
+        "read entitites",
+        "modify entities",
+        "read extracts",
+        "read outgoing-feeds",
+    ]
+    ids_required_for_user = []
+    for value in permissions_data:
+        if value["name"] in wanted_permissions:
+            ids_required_for_user.append(value["id"])
+
+    return ids_required_for_user
+
+
+def authenticate_user(ids_of_user: list, ids_required_for_user: list) -> Tuple[bool, List[int]]:
+    """Get user authentication and missing permission ids .
+    :param ids_of_user: permission ids user have
+    :type ids_of_user: list
+    :param ids_required_for_user: permission ids required for user to authenticate
+    :type ids_required_for_user: list
+    :return: is user authenticated , missing permissions ids
+    :rtype: boolean,list
+    """
+    user_authenticated = False
+    value = list(set(ids_required_for_user).difference(ids_of_user))
+
+    if not value:
+        user_authenticated = True
+    return user_authenticated, value
+
+
+def get_permission_name_from_id(permission_data: dict, permission_ids: list) -> Any:
+    """Get permission name from permission ids.
+    :param permission_data: permission data
+    :type permission_data: dict
+    :param permission_ids: permission id for authenticate
+    :type permission_ids: list
+    :return: permissions name
+    :rtype: Any
+    """
+    permissions_name = []
+    for data in permission_data:
+        for permission_id in permission_ids:
+            if data["id"] == permission_id:
+                permissions_name.append(data["name"])
+    return permissions_name
+
+
+def test_module(client: Client) -> Any:
+    """Tests API connectivity and authentication'
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+    :type client: ``Client``
+    :param Client: EclecticIQ client to use
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``Any``
+    """
 
+    # INTEGRATION DEVELOPER TIP
+    # Client class should raise the exceptions, but if the test fails
+    # the exception text is printed to the Cortex XSOAR UI.
+    # If you have some specific errors you want to capture (i.e. auth failure)
+    # you should catch the exception here and return a string with a more
+    # readable output (for example return 'Authentication Error, API Key
+    # invalid').
+    # Cortex XSOAR will print everything you return different than 'ok' as
+    # an error
+    try:
+        permissions_of_user = client.get_user_granted_permissions()
+    except Exception:
+        permissions_of_user = []
```

Comment:
What is it used for?

Reply:
'permissions_of_user' has been removed as it is not required, thanks for pointing out.